dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -112,21 +107,12 @@ private async Task<string> RenderAsync(List<Block> blocks, Kernel kernel, Kernel
                     break;
 
                 default:
-                    const string Error = "Unexpected block type, the block doesn't have a rendering method";
-                    this._logger.LogError(Error);
-                    throw new KernelException(Error);
+                    Debug.Fail($"Unexpected block type {block?.GetType()}, the block doesn't have a rendering method");
+                    break;
```

Comment:
Is `Debug.Fail` accidental change?
```suggestion
                    const string Error = "Unexpected block type, the block doesn't have a rendering method";
                    this._logger.LogError(Error);
                    throw new KernelException(Error);
```

Reply:
No, it's on purpose. This isn't reachable if the code is correct, right?

Reply:
Yeah, just wondering if we should throw an exception, in case we introduce incorrect code in the future. Looks like `Debug.Fail` causes the debugger to break at the line of code, so I was double-checking if this is something that could impact user experience.